server/src/main/java/org/opensearch/extensions/ExtensionsSettings.java

```diff
@@ -127,6 +130,10 @@ public List<ExtensionDependency> getDependencies() {
             return dependencies;
         }
 
+        public ExtensionAdditionalSettings getAdditionalSettings() {
```

Comment:
```suggestion
        public ExtensionAdditionalSettings getSettings() {
```
?

Reply:
I'm wondering if this whole class could be extension scoped settings. I added `additionalSettings` here after looking at the [SettingsModule](https://github.com/opensearch-project/OpenSearch/blob/main/server/src/main/java/org/opensearch/common/settings/SettingsModule.java#L73-L75) and seeing that its constructor took in additional settings for settings defined by plugins. From security's POV it needs access to the settings placed in `extensions.yml` that are pertinent to security and this `getAdditionalSettings` would provide those ExtensionScopedSettings so if the values were absent then security could still get and define their default values.